avocado/core/job.py

```diff
@@ -514,8 +510,27 @@ def _run(self):
         if 'FAIL' in summary:
             self.exitcode |= exit_codes.AVOCADO_TESTS_FAIL
 
+        self._post_run()
```

Comment:
I think it's about time we define how much we bundle into `Job.run()`, and properly define the other steps. Right now we have the handled `Job.run()`, and the unhandled `Job._run()` method, but there are many action embbeded in them, such as creating the test suite, running the pre/post scripts, etc.

I'll post a proposal for a more granular approach, and then we can decide if `Job` will have a all-encompassing `super_run()` (that would include test suite creation, pre job, test execution, post job, etc) or if that would be to the implementation of the runner itself (say the `run` core plugin).